client/lib/feed-stream-store/index.js

```diff
@@ -45,6 +45,11 @@ function mixedKeyMaker( post ) {
 	return siteKeyMaker( post );
 }
 
+const recommendedKeyMaker = post => ( {
+	...siteKeyMaker( post ),
+	isRecommendation: true }
```

Comment:
I see a feed-stream and a ui-stream as different things. 

A feed-stream is an ordered collection of posts, keyed off the post key in feed-stream-store, and likely keyed off the global_id in redux. A feed-stream can also have gaps, which are ranges in the stream where there are more posts we have no fetched yet. The head and tail of a feed-stream can be thought of as a gap too, though we didn't model it that way. 

ui-streams are an ordered collection of descriptions of ui elements. The reader/stream component is responsible for taking in a feed-stream and transforming that into a ui-stream. That started out as a 1:1 mapping, postKeys served double duty, both as the key for the feed-stream and for the ui-stream.

Now that's we've added fancier things to the ui-stream, and since redux will likely do away with the postKey in feed-stream-store, I think we need to make this separation more deliberate.  

This PR is reconsolidate the separation back into the postKey, which is the opposite direction I've been trying to go.